### PR TITLE
docs: Deploy docs site with Vercel

### PR DESCRIPTION
**Pull request by lorensr**

### CODE REVIEW

File: packages/docs/docusaurus.config.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const watch = ['y', 'yes', 't', 'true', '1'].includes(process.env.TYPEDOC_WATCH)
 module.exports = {
   title: 'Temporal Node.js SDK API Reference',
```
Comment (bergundy) [inline packages/docs/docusaurus.config.js:9]: May as well fix the title, I've been wondering where this title is coming from...

File: vercel.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "github": {
+    "silent": true
```
Comment (bergundy) [inline vercel.json:3]: What does this do?

**Review by bergundy — APPROVED** [review]

File: vercel.json (hunk 2 above)
Comment (lorensr) [reply vercel.json]: Removed, thanks—only relevant to the setup where Vercel does the docs build on each PR and push to main. (It stops the Vercel bot from commenting on every PR)

**Comment by lorensr** [comment]
Deploy prod docs from main worked:

image
image

**Pull request merged**